azure-mgmt-monitor/src/main/java/com/microsoft/azure/management/monitor/MetricDefinitions.java

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.management.monitor;
+
+import com.microsoft.azure.management.monitor.implementation.MonitorManagementClientImpl;
+import com.microsoft.azure.management.monitor.implementation.MonitorManager;
+import com.microsoft.azure.management.resources.fluentcore.arm.models.HasManager;
+import com.microsoft.azure.management.resources.fluentcore.model.HasInner;
+import rx.Observable;
+
+import java.util.List;
+
+
+/**
+ * Entry point for Monitor Metric Definitions API.
+ */
+public interface MetricDefinitions extends
+        HasManager<MonitorManager>,
+        HasInner<MonitorManagementClientImpl> {
+
+    /**
+     * Lists Metric Definitions for a given resource.
+     *
+     * @param resourceId The resource Id.
+     * @return list of metric definitions.
+     */
+    List<MetricDefinition> listByResource(String resourceId);
```

Comment:
List<QueryKey> queryKeys = new ArrayList<>();

    List<QueryKeyInner> queryKeyInners = this.manager().inner().queryKeys().listBySearchService(this.resourceGroupName(), this.name());
    if (queryKeyInners != null) {
      for (QueryKeyInner queryKeyInner : queryKeyInners) {
        queryKeys.add(new QueryKeyImpl(queryKeyInner));
      }
    }

    return Collections.unmodifiableList(queryKeys);

Reply:
fixed